src/players/FilePlayer.js

```diff
@@ -38,11 +38,13 @@ export default class FilePlayer extends Component {
       this.props.url !== prevProps.url &&
       !isMediaStream(this.props.url)
     ) {
+      this.player.src = ''
```

Comment:
I don’t think this is needed? This runs when the `url` has changed and the new one is not a MediaStream, we don’t necessarily want to clear the `src` here, especially if the new `url` is also a file path?

Reply:
@cookpete I have tested it without this line and it doesn't really fix the problem (old video still load when I change the URL)

I don't see any downside of clearing `src` here, as it'll be set again in the next render.

Reply:
>I don't see any downside of clearing src here, as it'll be set again in the next render.

Will it? This is in `componentDidUpdate` which fires after everything has finished rendering. When does the "next render" happen?

Reply:
@cookpete Oh you're right. Sorry, I was blind and somehow thought it is `componentWillUpdate`... Now, I don't even know why it's still working when I have this. 

I undo this change, and looks like reseting the src inside `componentWillUnmount` is enough in my case (as I actually unmount the videoPlayer)

Thanks for looking into this!